benchmark/Streamly/Benchmark/Data/Array.hs

```diff
@@ -4,99 +4,191 @@
 --
 -- License     : BSD-3-Clause
 -- Maintainer  : streamly@composewell.com
--- Stability   : experimental
--- Portability : GHC
 
 {-# LANGUAGE CPP #-}
 
-module Main (main) where
+import Control.DeepSeq (NFData(..))
+
+#ifndef DATA_PRIM_ARRAY
+import Control.DeepSeq (deepseq)
+#endif
 
-import Control.DeepSeq (NFData(..), deepseq)
 import System.Random (randomRIO)
 
+#ifdef MEMORY_ARRAY
+import qualified GHC.Exts as GHC
+#endif
+
 import qualified Streamly.Benchmark.Data.ArrayOps as Ops
-import qualified Streamly.Internal.Data.Array as A
-import qualified Streamly.Prelude as S
 
-import Streamly.Benchmark.Common
+import Streamly.Benchmark.Common hiding (benchPureSrc)
 import Gauge
 
+#ifdef MEMORY_ARRAY
+import Foreign.Storable (Storable(..))
+#endif
+
+#ifdef DATA_PRIM_ARRAY
```

Comment:
You could combine all imports for DATA_PRIM_ARRAY in a single conditional, and similarly for MEMORY_ARRAY.